### DISCUSSION TITLE
What is the most reliable and well-suited method for synchronizing MySQL tables with ClickHouse in a production environment?

### DISCUSSION THREAD

**Discussion by dotslash21** [ClickHouse/ClickHouse#54168] [Q&A] [answered]
We use a MySQL instance for backend services with numerous transactional queries. Our aim is to synchronize these tables with ClickHouse for both internal and external analytical queries. We've explored various methods:

1) **MaterializedMySQL**: Currently experimental.
2) **Kafka Connect & Debezium**: Popular and stable, yet challenging to configure and maintain.
3) **Kafka Connect & Debezium with Altinity ClickHouse sink**: Simpler setup, but potential issues with stability and support.
4) **Altinity Replicator for ClickHouse**: Easiest to implement, but with similar concerns as option 3.

We're leaning towards option 2 for its proven stability and support. Do you folks have any insights or alternative suggestions?

**Comment by arisarose** [comment]
@dotslash21,
I'm a TPM with ClickHouse. Is this an on-premise or cloud setup? 
The recommendation from our engineering team would be to stick with MaterializedMySQL. Although still experimental, you can use it and let us know of any  issues you may be having so we can make it better :)

Thanks!

**Reply by UnamedRus** [reply, marked as answer]
I would advice against MaterializedMySQL, because core (ClickHouse Inc) team doesn't support it, unlike MaterializedPostgreSQL

https://github.com/ClickHouse/ClickHouse/issues/32513#issuecomment-1579859302
https://github.com/ClickHouse/ClickHouse/issues/47821

> Kafka Connect & Debezium with Altinity ClickHouse sink: Simpler setup, but potential issues with stability and support.

Our connector in production use by several of our clients
